packages/typespec-vscode/test/extension/common/common-steps.ts

```diff
@@ -136,3 +137,24 @@ export function restoreTspConfigFile(folderName: string, lines: string) {
   const filePath = path.join(folderName, "tspconfig.yaml");
   fs.writeFileSync(filePath, lines, "utf-8");
 }
+
+/**
+ * Attempt to install the specified npm package and handle the error of pnpm version mismatch.
+ * @param pkg package name
+ * @returns Return true to skip testing, otherwise return false.
+ */
+export function tryInstallAndHandle(pkg: string): boolean {
+  try {
+    execSync(`pnpm install ${pkg}`, { stdio: "pipe" });
```

Comment:
where is this being run? I think the source of most of this is that we should not also be patching anything on the vscode package. 

Can this be done in a temp folder and link the lcoal version of every package(they are all there) to not depend on any extra remote issues